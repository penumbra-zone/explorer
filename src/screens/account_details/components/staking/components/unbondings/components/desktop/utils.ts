export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'validator',
    width: 25,
  },
  {
    key: 'commission',
    align: 'right',
    width: 25,
  },
  {
    key: 'amount',
    width: 25,
    align: 'right',
  },
  {
    key: 'linkedUntil',
    width: 25,
    align: 'right',
  },
];
