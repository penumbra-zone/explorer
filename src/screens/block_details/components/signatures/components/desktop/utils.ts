export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'validator',
    width: 34,
  },
  {
    key: 'votingPower',
    align: 'right',
    width: 33,
  },
  {
    key: 'signed',
    width: 33,
  },
];
