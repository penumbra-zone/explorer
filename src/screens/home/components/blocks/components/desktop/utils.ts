export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
}[] = [
  {
    key: 'height',
  },
  {
    key: 'validator',
  },
  {
    key: 'hash',
  },
  {
    key: 'txs',
  },
  {
    key: 'time',
    align: 'right',
  },
];
