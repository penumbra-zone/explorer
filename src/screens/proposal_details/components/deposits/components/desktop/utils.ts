export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'depositor',
    width: 34,
  },
  {
    key: 'amount',
    width: 33,
    align: 'right',
  },
  {
    key: 'time',
    width: 33,
  },
];
