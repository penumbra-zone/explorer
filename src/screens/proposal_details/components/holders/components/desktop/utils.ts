export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'rank',
    width: 10,
  },
  {
    key: 'address',
    width: 30,
  },
  {
    key: 'quantity',
    width: 20,
    align: 'right',
  },
  {
    key: 'percentage',
    width: 20,
  },
  {
    key: 'value',
    align: 'right',
    width: 20,
  },
];
