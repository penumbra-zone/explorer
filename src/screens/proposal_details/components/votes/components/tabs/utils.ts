export const tabLabels = [
  {
    key: 'all',
    num: 30,
  },
  {
    key: 'yes',
    num: 3,
  },
  {
    key: 'no',
    num: 3,
  },
  {
    key: 'noWithVeto',
    num: 3,
  },
];
