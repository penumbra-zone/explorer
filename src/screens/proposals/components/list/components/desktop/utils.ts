export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'idx',
    width: 15,
  },
  {
    key: 'token',
    width: 10,
  },
  {
    key: 'price',
    align: 'right',
    width: 15,
  },
  {
    key: 'change',
    align: 'right',
    width: 15,
  },
  {
    key: 'volume',
    align: 'right',
    width: 15,
  },
  {
    key: 'marketCap',
    align: 'right',
    width: 15,
  },
  {
    key: 'holders',
    align: 'right',
    width: 15,
  },
];
