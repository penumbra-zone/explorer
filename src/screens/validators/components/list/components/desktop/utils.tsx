import React from 'react';
import {
  Typography,
} from '@material-ui/core';
import { InfoPopover } from '@components';

export const fetchColumns = (t): {
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
  component?: React.ReactNode;
  sort?: boolean;
}[] => {
  return ([
    {
      key: 'idx',
      width: 10,
    },
    {
      key: 'validator',
      width: 20,
      sort: true,
    },
    {
      key: 'stake',
      align: 'right',
      width: 15,
      sort: true,
    },
    {
      key: 'fee',
      align: 'right',
      width: 10,
      sort: true,
    },
    {
      key: 'lastVote',
      align: 'right',
      width: 15,
    },
    {
      key: 'skipRate',
      width: 15,
      sort: true,
    },
    {
      key: 'condition',
      align: 'center',
      width: 15,
      component: (
        <Typography variant="h4" className="label popover">
          {t('condition')}
          <InfoPopover
            content={t('conditionExplanation')}
          />
        </Typography>
      ),
    },
  ]);
};
