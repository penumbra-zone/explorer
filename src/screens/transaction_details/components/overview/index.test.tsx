import React from 'react';
import renderer from 'react-test-renderer';
import { MockTheme } from '@tests/utils';
import Overview from '.';

// ==================================
// mocks
// ==================================
jest.mock('@components', () => ({
  BoxDetails: (props) => <div id="BoxDetails" {...props} />,
  Result: (props) => <div id="Result" {...props} />,
}));

jest.mock('dayjs', () => {
  const mockTest = () => ({
    format: jest.fn(() => '2020-08-10 12:00:00'),
  });
  return mockTest;
});

// ==================================
// unit tests
// ==================================
describe('screen: BlockDetails/Overview', () => {
  it('matches snapshot', () => {
    const component = renderer.create(
      <MockTheme>
        <Overview />
      </MockTheme>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
