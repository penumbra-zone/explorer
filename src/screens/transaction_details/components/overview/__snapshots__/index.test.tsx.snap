// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen: BlockDetails/Overview matches snapshot 1`] = `
<div
  details={
    Array [
      Object {
        "detail": "4SGxuRMcseNbwki3tGxXPpfz7iFnuo9FUpTfiM4gJ8rhH59uZYSBBK2zW27xRdGX8Sb2N4VkGUnBYt59SBKEhPfB",
        "label": "transactions:hash",
      },
      Object {
        "detail": <Link
          href="/blocks/123"
          passHref={true}
        >
          <WithStyles(ForwardRef(Typography))
            className="value"
            component="a"
            variant="body1"
          >
            812,768,640
          </WithStyles(ForwardRef(Typography))>
        </Link>,
        "label": "transactions:height",
      },
      Object {
        "detail": "2020-08-10 12:00:00",
        "label": "transactions:time",
      },
      Object {
        "detail": "123 udaric",
        "label": "transactions:fee",
      },
      Object {
        "detail": "100 / 200",
        "label": "transactions:gas",
      },
      Object {
        "detail": <Result
          success={true}
        />,
        "label": "transactions:result",
      },
      Object {
        "className": "makeStyles-memo",
        "detail": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam vel cursus tortor. Fusce lobortis sollicitudin dolor id mollis. Nullam quam ex, dignissim eu eros vel, tincidunt ultrices ligula.",
        "label": "transactions:memo",
      },
    ]
  }
  id="BoxDetails"
  title="transactions:overview"
/>
`;
